cordyceps/src/list.rs

```diff
@@ -963,7 +963,7 @@ impl<T: Linked<Links<T>> + ?Sized> List<T> {
         let start_links = T::links(splice_start).as_mut();
         let end_links = T::links(splice_end).as_mut();
         debug_assert!(
-            splice_start == splice_end
+            core::ptr::addr_eq(splice_start.as_ptr(), splice_end.as_ptr())
```

Comment:
similarly, we could write:
```suggestion
            ptr::addr_eq(splice_start.as_ptr(), splice_end.as_ptr())
```